### PR TITLE
Add checks to version updating code

### LINKED ISSUE #55: Refactor version detection
We should detect the type of project, then look for only the old version for the file where it's expected to exist. The current script happens to work for our projects because of how it recurses through the tree structure and the order of files it checks for, but this is unreliable

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #55

#### What's this PR do?
 - removes support for versions in `setup.py`. I believe all of our projects now put the version in an `__init__.py` file and reference it from `setup.py`.

#### How should this be manually tested?
Do a release

### CODE REVIEW

**Comment by annagav** [comment]
This is only for doof to use, right? I think there are some old projects for which the version could be found in various inconsistent places, so the release script should still work for those...

**Comment by annagav** [comment]
This is only for doof to use, right? I think there are some old projects for which the version could be found in various inconsistent places, so the release script should still work for those...

File: README.md [hunk 1]
```diff
@@ -143,7 +143,7 @@ If the test works, upload to the real deal:
 3.  The ``util`` directory contains the templates to format the release notes
     and the GitHub descriptions.
 4.  The script expects to find the current release number in either the Django
-    ``settings.py`` file or in the ``setup.py`` file in the project root
+    ``settings.py`` file or in the ``__init__.py`` file in the project root
```
Comment (annagav) [inline README.md]: So we have two different scripts now `release.sh` and `release.py`. Do the still have the same behavior? It is very unclear in the documentation which one we talking about.

Comment (noisecapella) [reply README.md]: They have the same behavior. `release.sh` will just call `release.py` and look up the repo url for the repo directory to pass it as an argument.

**Comment by noisecapella** [comment]
I'll add back the `setup.py` check

**Comment by annagav** [comment]
I think it looks for the version in `setup.py` even if it found it `__init__.py`

**Comment by noisecapella** [comment]
I fixed the problem, can you try it again?

**Pull request merged**